zcash_client_backend/Cargo.toml

```diff
@@ -17,29 +17,55 @@ rust-version = "1.56.1"
 development = ["zcash_proofs"]
 
 [dependencies]
+zcash_address = { version = "0.2", path = "../components/zcash_address" }
+zcash_encoding = { version = "0.2", path = "../components/zcash_encoding" }
+zcash_note_encryption = { version = "0.2", path = "../components/zcash_note_encryption" }
+zcash_primitives = { version = "0.8", path = "../zcash_primitives" }
+
+# Dependencies exposed in a public API:
+# (Breaking upgrades to these require a breaking upgrade to this crate.)
+# - Data Access API
+time = "0.2"
+
+# - Encodings
 base64 = "0.13"
 bech32 = "0.8"
-bls12_381 = "0.7"
 bs58 = { version = "0.4", features = ["check"] }
-byteorder = { version = "1", optional = true }
-crossbeam-channel = "0.5"
-group = "0.12"
+
+# - Errors
 hdwallet = { version = "0.3.1", optional = true }
+
+# - Logging and metrics
 memuse = "0.2"
-nom = "7"
-orchard = "0.3"
-percent-encoding = "2.1.0"
-proptest = { version = "1.0.0", optional = true }
+tracing = "0.1"
+
+# - Protobuf interfaces
 protobuf = "~2.27.1" # MSRV 1.52.1
-rayon = "1.5"
+
+# - Secret management
 secrecy = "0.8"
 subtle = "2.2.3"
-time = "0.2"
-tracing = "0.1"
-zcash_address = { version = "0.2", path = "../components/zcash_address" }
-zcash_encoding = { version = "0.2", path = "../components/zcash_encoding" }
-zcash_note_encryption = { version = "0.2", path = "../components/zcash_note_encryption" }
-zcash_primitives = { version = "0.8", path = "../zcash_primitives" }
+
+# - Shielded protocols
+bls12_381 = "0.7"
+group = "0.12"
+orchard = "0.3"
+
+# - Test dependencies
+proptest = { version = "1.0.0", optional = true }
```

Comment:
Should this be a dev dependency?

Reply:
This is included as a normal dependency because we expose it in the `testing` modules under the `test-dependency` feature flags, so that if someone needs proptest generators for their own tests, they have them.